docs/concepts.rst

```diff
@@ -842,61 +842,135 @@ values by "case" instead of by entity.
+each other.
 
 .. code-block:: python
 
     builder.set("core__parallel_execution__enabled", True)
 
 When parallel execution is enabled, Bionic starts up several worker processes
-[#workers]_, each of which can work on one value at a time. Of course, a worker can
-only start computing a value once all its dependencies are complete, so the number of
-processes that can be working at once depends on the :ref:`dependency graph
-<dagviz>`: if there aren't many branches in the graph, then most of the processes
-won't do much work. It does take extra time to set up the processes and move
-information between them, so parallel execution is not guaranteed to be faster
-overall. However, in general, if you have many expensive operations which don't
-depend on each other, enabling parallelism will improve performance.
-
-By default, Bionic will create one worker process for each CPU on your machine. This
-is usually a sensible number, but it can also be set directly:
+[#workers]_, each of which can work on one value at a time. By default, Bionic
+will create one worker process for each CPU on your machine. This is usually a
+sensible number, but it can also be set directly:
 
 .. code-block:: python
 
     builder.set("core__parallel_execution__worker_count", 8)
 
+.. [#workers] The pool of workers is managed by
+  `Loky <https://loky.readthedocs.io/en/stable/>`_,
+  which is built on Python's
+  `multiprocessing <https://docs.python.org/3.8/library/multiprocessing.html>`_ module.
+  The pool is global and reusable, so it should only need to be initialized once in
+  the lifetime of the main process.
+
+Google AI Platform Execution
+............................
+
+.. versionadded:: 0.9.0
+
+Bionic can make use of compute resources in the cloud by sending entities to
+`Google AI Platform (AIP) <https://cloud.google.com/ai-platform>`_ for
+computation.
+
+Using this feature requires specifying the GCP project as well as a docker image
+in Google Container Registry containing the same Python environment (runtime and
+libraries) as the one used to invoke Bionic.
+
+.. code-block:: python
+
+    builder.set('core__aip_execution__enabled', True)
+    builder.set('core__aip_execution__gcp_project_name', 'my-gcp-project')
+    builder.set('core__aip_execution__docker_image_name', 'bionic:latest')
+
+Additionally, :ref:`Google Cloud Storage <google_cloud_storage_anchor>` must be
+enabled.
+
+Entities marked with :func:`@aip_task_config <bionic.aip_task_config>` decorator
+will use AIP for computation. It takes in a `Compute Engine machine type identifier <https://cloud.google.com/ai-platform/training/docs/machine-types#compute-engine-machine-types>`_.
+Such entities must be serializable and cannot be marked with :func:`@persist(False) <bionic.persist>`.
+
+.. code-block:: python
+
+    @builder
+    @bionic.aip_task_config('n1-standard-4')
+    def x():
+      return 1
+
+Please refer to Docker documentation on how to create a docker image and push
```

Comment:
There are multiple paragraphs between this one and the one above where we mention Docker images, so perhaps this should be a named section that we link to from the paragraph above?